src/main/kotlin/com/the_chance/endpoints/Categories.kt

```diff
@@ -0,0 +1,84 @@
+package com.the_chance.endpoints
+
+import com.the_chance.data.ServerResponse
+import com.the_chance.data.services.CategoryService
+import com.the_chance.utils.toLowerCase
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import java.lang.Exception
+
+fun Route.categoryRoutes(categoryService: CategoryService) {
+
+    get("/categories") {
+        val categories = categoryService.getAllCategories()
+        call.respond(ServerResponse.success(categories))
+    }
+
+    post("/category") {
+        val params = call.receiveParameters()
+        val categoryName = params["name"]?.trim().orEmpty()
+        val categoryImage = params["image"]?.trim()?.trim().orEmpty()
+        val categoryList = categoryService.getAllCategories()
+
+        if (categoryList.any { it.name.toLowerCase() == categoryName.toLowerCase() }) {
```

Comment:
this should be inside the function no need to be here.
and after checking the length of the category